File: src/app/interfaces/application.ts
import { ApplicationStates } from './application-states.enum';
import { BusinessUnitManager } from './business-unit-manager';
import { Note } from './note';
import { Applicant } from './applicant';
import { BusinessUnit } from './business-unit';
import { Region } from './region';

export interface Application {
  id: number;
  applicant: number | Applicant;
  businessUnitId?: number;
  businessUnitManagerId?: number;
  notes: Note[];
  regionId?: number;
  state: ApplicationStates;
  motivationLetterLink?: string;
  firstInterviewDateTime?: Date;
  secondInterviewDateTime?: Date;
  title: string;
  department: string;
}

File: recruitment-experience-core/src/main/java/nl/ordina/recruitmentexperience/core/model/Application.java
package nl.ordina.recruitmentexperience.core.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import nl.ordina.recruitmentexperience.core.model.state.State;

import java.time.OffsetDateTime;

@Getter
@Builder
public class Application {
    private Long id;
    private Applicant applicant;
    private BusinessUnit businessUnit;
    private BusinessUnitManager businessUnitManager;
    private Region region;
    @Setter
    private State state;
    private String motivationLetterLink;
    private String title;
    private OffsetDateTime firstInterviewDateTime;
    private OffsetDateTime secondInterviewDateTime;
    private Department department;
}
